### PR TITLE
Create protocol-specs to verify transactionRoot of blocks - Closes #5208

### LINKED ISSUE #5208: Create protocol-specs to verify transactionRoot of blocks
### Description

Protocol specifications require a `transactionRoot` property of blocks, which is the merkle root of transaction IDs in a block.

### Motivation

Proper protocol specifications are required for testing merkle root functionality. 

### Acceptance Criteria

Generator which outputs correct data.

### Additional information

Related issues: #5204

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?

This PR resolves #5208 

### How was it solved?

- Created protocol specification for `merkleRoot` calculation

### How was it tested?

- Run `npm run generate-all-specs` to get the transaction merkle root.

### CODE REVIEW

File: protocol-specs/generators/transaction_merkle_root_for_blocks/index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+const generateTransactionMerkleRoot = () => {
+	const transactionIds = getRandomTransactionIds();
+	const ids = transactionIds.map(t => bufferToHex(t));
+	const transactionMerkleRoot = merkleRoot(transactionIds);
+
+	return {
+		description: 'Given a valid block',
+		input: {
+			transactionIds: ids,
```
Comment (shuse2) [inline protocol-specs/generators/transaction_merkle_root_for_blocks/index.js:61]: can we make multiple cases for the different `transactionIds` size?
Possibly,  `[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 50, 150, 300]` something like this?

File: protocol-specs/generators/transaction_merkle_root_for_blocks/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+const concat = (m1, m2) => Buffer.concat([m1, m2]);
+
+const leafHash = m => hash(concat(LEAFPREFIX, m));
+
+const branchHash = m => hash(concat(BRANCHPREFIX, m));
+
+const merkleRoot = transactionIds => {
+	const len = transactionIds.length;
+	if (len === 0) return Buffer.from([]);
```
Comment (ricott1) [inline protocol-specs/generators/transaction_merkle_root_for_blocks/index.js:44]: should return emptyHash
```suggestion
	if (len === 0) return hash(Buffer.from([]));
```

File: protocol-specs/generators/transaction_merkle_root_for_blocks/index.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	const k = 2 ** Math.floor(Math.log2(len - 1));
+	const newData = transactionIds.splice(k);
+
+	return branchHash(concat(merkleRoot(transactionIds), merkleRoot(newData)));
+};
+
+const generateTransactionMerkleRoot = () =>
+	[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 50, 150, 300].map(count => {
+		const transactionIds = getRandomTransactionIds(count);
+		const ids = transactionIds.map(t => bufferToHex(t));
```
Comment (ricott1) [inline protocol-specs/generators/transaction_merkle_root_for_blocks/index.js:56]: why do you convert to hex here? couldnt we just work with buffers?

File: protocol-specs/generators/transaction_merkle_root_for_blocks/index.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+		const transactionMerkleRoot = merkleRoot(transactionIds);
```
Comment (ricott1) [inline protocol-specs/generators/transaction_merkle_root_for_blocks/index.js:57]: it looks to me that indeed `merkleRoot` should receive an array of buffers, maybe im misunderstanding what `bufferToHex` does

**Pull request merged**